### PR TITLE
Openrtb compliant Prebid Server Adapter

### PR DESCRIPTION
**Pull request by anwzhang**
This PR includes changes that works with latest Prebid Server openrtb endpoint.

Note that this is tested only with local test server, haven't tested with https request (SSL creatives), and cache function for MoPub builds.

Actual endpoint switch will happen later when Prebid Server fully rolls out its open rtb endpoint. If we use endpoint from AppNexus in the prebid server adapter, we still anticipate the OPEN RTB protocol stays the same.

### CODE REVIEW

File: PrebidMobile/PrebidMobileCore/src/main/java/org/prebid/mobile/core/Prebid.java [hunk 1]
```diff
@@ -424,7 +429,7 @@ public static boolean isSecureConnection() {
      */
     public static void shouldLoadOverSecureConnection(boolean secureConnection) {
         // Only enables overrides for MoPub, DFP should always load over secured connection
-        if (getClassFromString(MOPUB_ADVIEW_CLASS) != null) {
+        if (Prebid.adServer.equals(AdServer.MOPUB)) {
```
[the remark this answers is not in the document]
Comment (anwzhang) [reply PrebidMobile/PrebidMobileCore/src/main/java/org/prebid/mobile/core/Prebid.java]: This is the same implementation as yours. If another adserver comes in and allow https only, you would have to add a check too.

File: PrebidMobile/PrebidServerAdapter/src/main/java/org/prebid/mobile/prebidserver/PrebidServerAdapter.java [hunk 2]
```diff
@@ -59,82 +58,62 @@ public void onServerResponded(JSONObject response) {
...
+                                    if (adUnit != null) {
+                                        final double bidPrice = bid.getDouble("price");
+                                        ArrayList<BidResponse> responseList = responses.get(adUnit);
+                                        if (responseList == null) {
+                                            responseList = new ArrayList<BidResponse>();
+                                        }
+                                        BidResponse newBid;
+                                        JSONObject targetingKeywords = bid.getJSONObject("ext").getJSONObject("prebid").getJSONObject("targeting");
+                                        if (Prebid.getAdServer() == Prebid.AdServer.DFP) {
+                                            String format = targetingKeywords.getString("hb_creative_loadtype");
```
[the remark this answers is not in the document]
Comment (anwzhang) [reply PrebidMobile/PrebidServerAdapter/src/main/java/org/prebid/mobile/prebidserver/PrebidServerAdapter.java]: Android uses a different way of caching, it needs format for know where to retrieve the cache object.

[the remark this answers is not in the document]
Comment (ppuviarasu) [reply PrebidMobile/PrebidServerAdapter/src/main/java/org/prebid/mobile/prebidserver/PrebidServerAdapter.java]: 👍

File: PrebidMobile/PrebidMobileCore/src/main/java/org/prebid/mobile/core/Prebid.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (ppuviarasu) [reply PrebidMobile/PrebidMobileCore/src/main/java/org/prebid/mobile/core/Prebid.java]: 👍

File: PrebidMobile/PrebidServerAdapter/src/main/java/org/prebid/mobile/prebidserver/PrebidServerAdapter.java [hunk 3]
```diff
@@ -59,82 +58,62 @@ public void onServerResponded(JSONObject response) {
...
+                                    AdUnit adUnit = getAdUnitByCode(impId);
+                                    if (adUnit != null) {
+                                        final double bidPrice = bid.getDouble("price");
+                                        ArrayList<BidResponse> responseList = responses.get(adUnit);
+                                        if (responseList == null) {
+                                            responseList = new ArrayList<BidResponse>();
+                                        }
+                                        BidResponse newBid;
+                                        JSONObject targetingKeywords = bid.getJSONObject("ext").getJSONObject("prebid").getJSONObject("targeting");
+                                        if (Prebid.getAdServer() == Prebid.AdServer.DFP) {
```
[the remark this answers is not in the document]
Comment (anwzhang) [reply PrebidMobile/PrebidServerAdapter/src/main/java/org/prebid/mobile/prebidserver/PrebidServerAdapter.java]: Done

[the remark this answers is not in the document]
Comment (ppuviarasu) [reply PrebidMobile/PrebidServerAdapter/src/main/java/org/prebid/mobile/prebidserver/PrebidServerAdapter.java]: 👍

**Pull request merged**